plugins/modules/selinux.py

```diff
@@ -119,6 +120,34 @@ def get_config_policy(configfile):
             return line.split('=')[1].strip()
 
 
+def get_kernel_enabled(module, grubby_bin):
+    if grubby_bin is None:
+        module.warn("'grubby' command not found on host - kernel boot configuration won't be updated")
```

Comment:
In my opinion, this breaks Ansible design at some point: kernel boot params are very important (they're able to override settings in /etc) and should not be treated with a warning. Either the module is able to change a kernel boot parameter, either it is not, and in the last case the module should fail.

It would probably be better to add a module option (boolean ?) to control this. So if, say, `update_bootloader` is *true*, and grubby is not available (because of the distribution - say, debian - or the architecture - say, arm64), the module fails.

Warning the user assumes that he conscientiously reads logs even when the playbook succeeds, whereas failing the task tells the user that the expected result is not reachable this way. Maybe I'm wrong, but that sounds more "ansible" for me.

Reply:
Good point - I'll add the boolean option as you are proposing.